definitions/features/katello_service.rb

```diff
@@ -27,13 +29,65 @@ def make_start(spinner, options = {})
       spinner.update 'No katello service to start'
     else
       filters = "--only #{services.join(',')}"
-      spinner.update 'Starting the katello services..'
+      spinner.update 'Starting katello services..'
       execute!("katello-service start #{filters}")
     end
   end
 
+  def restart(spinner, options = {})
+    spinner.update 'Restarting katello services..'
+    if options[:only] || options[:exclude]
+      filters = construct_filters_for_restart(options)
+      execute!("katello-service restart #{filters}")
+    else
+      execute!('katello-service restart')
+    end
+  end
+
+  def run_with_hammer_ping_retry(spinner, action = '', services_options = {})
+    meth_call_by_action(action, spinner, services_options)
+    spinner.update 'Sometimes, services will take time to be ready'
+    begin
+      retry_hammer_ping(spinner)
+    rescue StandardError => e
+      logger.error e.message
+    end
+  end
+
   private
 
+  def retry_hammer_ping(spinner)
+    RETRIES_FOR_SERVICES_RESTART.times do |retry_count|
+      spinner.update "Retry #{retry_count + 1}: checking status by hammer ping"
+      result = feature(:hammer).hammer_ping_cmd
+      if result[:success]
+        spinner.update 'All services are running.'
+        break
+      elsif retry_count < (RETRIES_FOR_SERVICES_RESTART - 1)
+        spinner.update 'Lets try checking status once again'
```

Comment:
What about some more descriptive message, such as (note the "\n" at the first line is needed for properly getting the message to the new line):

```
        puts "\n#{result[:message]}"
        spinner.update "Waiting #{PING_RETRY_INTERVAL} seconds before retry."
        sleep PING_RETRY_INTERVAL
```